app/services/motion_service.rb

```diff
@@ -36,27 +37,20 @@ def self.reopen(motion, close_at)
   end
 
   def self.close(motion)
-    motion.store_users_that_didnt_vote
-    motion.closed_at = Time.now
-    motion.save!
-    motion.update_members_not_voted_count
-
+    motion.close!
     EventBus.broadcast('motion_close', motion)
     Events::MotionClosed.publish!(motion)
   end
 
   def self.close_by_user(motion, user)
     user.ability.authorize! :close, motion
-
-    motion.store_users_that_didnt_vote
-    motion.closed_at = Time.now
-    motion.save!
-
+    motion.close!
     EventBus.broadcast('motion_close_by_user', motion, user)
     Events::MotionClosedByUser.publish!(motion, user)
   end
 
   def self.create_outcome(motion:, params:, actor:)
+    close(motion) if motion.needs_to_be_closed?
```

Comment:
I found myself thinking 'is it odd that an unauthorized user can close a lapsed motion by trying to create an outcome for it?'

But... meh. 💛